tests/unit/project_test.py

```diff
@@ -846,6 +848,23 @@ def test_avoid_multiple_push(self):
             project.push()
             assert fake_push.call_count == 2
 
+    def test_push_duplicate_image(self):
+        svc_no_build = Service(
+            'busy1', image=BUSYBOX_IMAGE_WITH_TAG,
+            client=self.mock_client)
+        svc_with_build = Service(
+            'busy2', image=BUSYBOX_IMAGE_WITH_TAG,
```

Comment:
See my comment about digests; can you add a test-case that has a service with a digest in the image reference?

Reply:
If we add config validation for "build + digest image", we won't need this check